src/hooks/helper/plasmUtils.ts

```diff
@@ -67,32 +68,13 @@ export const defaultAmountWithDecimals = (
   }
 };
 
-export const reduceDenomToBalance = (bal: number, unit: number, decimal: number) => {
-  const unit_decimal = unit + decimal;
-  const decPoint = new BN(10).pow(new BN(unit_decimal));
-  // console.log('d', decPoint.toString())
-
-  const strBal = bal.toString();
-  // console.log('b', bal.toString())
-
-  let formatted = new BN(0);
-
-  const arrDecimalBal = strBal.split('.');
-  if (arrDecimalBal.length === 2) {
-    const intBal = arrDecimalBal[0];
-    const minorityBal = arrDecimalBal[1];
-    const remainNum = Number(`${intBal}${minorityBal}`);
-    const decimalLength = minorityBal.length;
-    // console.log('intBal', intBal)
-    // console.log('minorityBal', minorityBal)
-    // console.log('decimalLength', decimalLength)
-    const divPoint = new BN(10).pow(new BN(decimalLength));
-    formatted = decPoint.mul(new BN(remainNum)).div(divPoint);
-  } else {
-    formatted = decPoint.mul(new BN(bal));
-  }
-  // console.log('f', formatted.toString())
-  return formatted;
+/**
+ * Convert the given value into the 18 decimals amount.
+ * @param amount 0.1
+ * @returns '100000000000000000' (format: BN)
+ */
+export const parseTo18Decimals = (amount: number | string): BN => {
```

Comment:
@niklabh Just an FYI. I would like to move this entire script to Astar.js library.

Reply:
ok